### PR TITLE
normalized python3 compatibility, shebang and encoding

### PR DESCRIPTION
**Pull request by ultrabug**
This series of patches ensures that all the Python code base is python3 compliant
and consistent by applying the following logic:

- python3 classifier on setup.py to explicitly state our python compatibility matrix
- add UTF-8 encoding header
- correct every shebang to the same /usr/bin/env python3
- shebang is only added on scripts meant to be executed on their own (removed otherwise)
- migrate some leftover scripts from python2 to python3 with minimal QA

This work is important to prepare for a more drastic change on Python code styling
using the black formatter and the setting up of automated QA checks on Python code base.

### CODE REVIEW

**Comment by ultrabug** [comment]
This PR is meant to help in the discussion: https://groups.google.com/forum/#!topic/scylladb-dev/14woftJDSaM

I will close it when we conclude there.

File: scylla-housekeeping [hunk 1]
```diff
@@ -1,4 +1,5 @@
-#!/usr/bin/python2
+#!/usr/bin/env python3
```
Comment (amoskong) [inline scylla-housekeeping]: Hi @ultrabug , this introduce a problem (only exists on CentOS, deb like distros are good): 

```
| [stderr] Traceback (most recent call last):
| [stderr]   File "/usr/lib/scylla/scylla-housekeeping", line 33, in <module>
| [stderr]     from pkg_resources import parse_version
| [stderr] ImportError: No module named 'pkg_resources'
```

I solved the problem by installing python34-setuptools-19.6.2-3.el7.noarch
I had asked @amnonh / @syuu1228 to update dist/redhat/scylla.spec.mustache.

Comment (amoskong) [reply scylla-housekeeping]: Sent a patch to scylladb-dev list:
[PATCH scylla] redhat/scylla.spec: add python34-setuptools dependency

**Comment by ultrabug** [comment]
Thanks a lot @amoskong 

Do you have any hint on how we could improve the tests suite to check for this maybe?

**Pull request merged**